src/components/Board/BoardShare/BoardShare.component.js

```diff
@@ -67,28 +74,36 @@ const BoardShare = ({
       </DialogTitle>
       <DialogContent className="ShareDialog__content">
         <div className="ShareDialog__Subtitle">
-          <div>{isOwnBoard && <Button
-            color="primary"
-            variant={isPublic ? "outlined" : "contained"}
-            className="ShareDialog__ToggleStatusButton"
-            onClick={publishBoard}
-            >
-              {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
-            </Button>
+          <div>{
+            isOwnBoard  
+              ? <Button
+              color="primary"
+              variant={isPublic ? "outlined" : "contained"}
+              className="ShareDialog__ToggleStatusButton"
+              onClick={publishBoard}
+              >
+                {!isPublic? <FormattedMessage {...messages.publishBoard } /> : <FormattedMessage {...messages.unpublishBoard } />}
+              </Button>
+              : <><Alert severity="warning">warning — you need to be registered to share a board</Alert>
+                <Button
+                  color="primary"
+                  variant="contained"
+                  onClick={OpenLogWd}
+                >LOGIN or SING UP</Button></>
```

Comment:
you have to use a string from the messages file in order to support translations